xgboost-job/xgboost-operator/overlays/application/kustomization.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```

Comment:
Application should not depend on base.
The application manifest is not overloading any resources defined in the base package.
So either

1) create a single package which includes the application resource
or
2) Create 2 separate packages one with the base resources and one with the application resource 
    and then create a 3rd package to update them.

Reply:
Currently other operators all follow this same pattern. We can fix them altogether in a follow-up PR.